ocean_lib/ocean/test/test_ocean_assets.py

```diff
@@ -237,18 +211,23 @@ def test_ocean_assets_algorithm(publisher_ocean_instance, publisher_wallet, conf
     assert ddo, "DDO None. The ddo is not cached after the creation."
 
 
-@pytest.mark.skip(reason="TODO: download function on OceanAssets class")
 def test_download_fails(publisher_ocean_instance, publisher_wallet):
     """Tests failures of assets download function."""
     with patch("ocean_lib.ocean.ocean_assets.OceanAssets.resolve") as mock:
-        mock.return_value = get_sample_ddo()
+        mock.return_value = Asset.from_dict(get_sample_ddo())
+        asset = mock.return_value
```

Comment:
please switch these two statements. Assign the asset first, then set it as the mock.return_value.